app/1.0.x/plugin-development/custom-entities.md

```diff
@@ -32,88 +36,99 @@ returns them.
 If you plugin is intended to support both Cassandra and PostgreSQL, then both
 migrations must be written.
 
-Each migration must bear a unique name, and `up` and `down` fields. Such fields
-can either be strings of SQL/CQL queries for simple migrations, or actual Lua
-code to execute for complex ones. The `up` field will be executed when Kong
-migrates **forward**. It must bring your database's schema to the latest state
-required by your plugin. The `down` field must execute the necessary actions to
-revert your schema to its previous state, before `up` was ran.
+If your plugin doesn't have it already, you should add a `<plugin_name>/migrations`
+folder to it. If there is no `init.lua` file inside. This is where all the migrations
+for your plugin will be referenced.
 
-One of the main benefits of this approach is should you need to release a new
-version of your plugin that modifies a model, you can add new migrations to the
-array before releasing your plugin. Another benefit is that it is also possible
-to revert such migrations.
+The initial version of your `migrations/init.lua` file will point to a single migration.
 
-As described in the [file structure]({{page.book.chapters.file-structure}})
-chapter, your migrations modules must be named:
+In this case we have called it `000_base`.
 
-```
-"kong.plugins.<plugin_name>.migrations.cassandra"
-"kong.plugins.<plugin_name>.migrations.postgres"
+``` lua
+-- `migrations/init.lua`
+return {
+  "000_base",
+}
 ```
 
-Here is an example of how one would define a migration file to store API keys:
+This means that there will be a file in `<plugin_name>/migrations/000_base.lua` containing the
+initial migrations. We'll see how this is done in a minute.
 
-```lua
--- cassandra.lua
+## Adding a new migration to an existing plugin
+
+Sometimes it is necessary to introduce changes after a version of a plugin has already been
+released. A new functionality might be needed. A database table row might need changing.
+
+When this happens, it is strongly recommended that you create a new migrations file instead
+of modifying the existing ones.
```

Comment:
not critical, but I'm concerned this indirectly suggests the wrong way as "a way" by making right way a "recommendation".

```
When this happens, it is strongly recommended that you create a new migrations file instead
of modifying the existing ones.
```
-> 
```
When this happens, you must create a new migrations file that performs this changes, so that users of your plugin can convert their existing databases to the new format. Migrations of past versions must never be modified.
```